screenpy/actions/either.py

```diff
@@ -0,0 +1,92 @@
+"""
+try/accept logic using screenplay pattern
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from screenpy.pacing import the_narrator
+from screenpy.speech_tools import get_additive_description
+
+if TYPE_CHECKING:
+    from screenpy import Actor
+    from screenpy.protocols import Performable
+
+
+class Either:
+    """Either perform followup action if the first one fails.
+
+    Allows actors to perform try/except blocks while still using screenplay pattern.
+    Examples::
+
+        the_actor.will(Either(DoAction()).or_(DoDifferentAction())
+
+        the_actor.will(
+            Either(DoAction()).otherwise(DoDifferentAction()).ignoring(
+                AssertionError, NotImplementedError
+            )
+        )
+
+        the_actor.will(
+            Either(CheckIfOnDomain(URL())).or_(Open.their_browser_on(URL())
+        )
+
+    """
+
+    except_performables: tuple[Performable, ...]
+    ignore_exceptions: tuple[type[BaseException], ...]
+
+    def perform_as(self, the_actor: Actor) -> None:
+        """perform a try/accept using the two provided actions"""
+        # logs the first attempt even if it fails
+        # try:
+        #     with the_narrator.mic_cable_kinked():
+        #         the_actor.will(*self.try_performables)
+        # except AssertionError:
+        #     try:
+        #         the_actor.will(*self.except_performables)
+        #     except Exception as exc:
+        #         raise exc from None
+
+        # Since we are fully expecting assertion error to be raised we kink the
+        # cable to avoid the explanation; which only happens when the cable is not
+        # kinked
+
+        # logs the first attempt only if it succeeds.
+        with the_narrator.mic_cable_kinked():
+            try:
+                the_actor.will(*self.try_performables)
+                return
+            except self.ignore_exceptions:
+                the_narrator.clear_backup()
+
+        the_actor.will(*self.except_performables)
+        return
+
+    def or_(self, *except_performables: Performable) -> Either:
+        """submit the except_performables action"""
```

Comment:
Remember that these docstrings show up in the auto-generated documentation, so it should follow the imperitive, short, descriptive first line and there should be args definition:

```python
"""Provide the alternative routine to perform.

Args:
...
"""
```
... actually i just looked through our other classes for examples to see how we have the other ones, and we don't actually follow this standard. I'm certain i've done this somewhere in here!

Maybe these are fine for now and i'll fix them when i do the Big Docstring Standardization i'm planning on doing.

Reply:
It should at least list the `Aliases` though!
```python
"""Provide the alternative routine to perform.

Aliases:
    * :meth:`~screenpy.actions.Either.except_`
    * :meth:`~screenpy.actions.Either.else_`
    * :meth:`~screenpy.actions.Either.otherwise`
    * :meth:`~screenpy.actions.Either.alternatively`
    * :meth:`~screenpy.actions.Either.failing_that`
"""
```

Reply:
Fixed